### ISSUE TITLE
Add `minPythonVersion`, `maxPythonVersion`, `black` and `flake8` to `PythonProject`

### ISSUE THREAD

**Issue by garysassano** [projen/projen#3379]
Add the following `NodeProject` equivalents to `PythonProject`:

- `minNodeVersion` (semver) → `minPythonVersion` (semver)
- `maxNodeVersion` (semver) → `maxPythonVersion` (semver)
- `prettier` (boolean) → `black` (boolean)
- `eslint` (boolean) → `flake8` (boolean)

**Comment by mrgrain** [comment]
Agree re `black` and `flake8`.

Not sure I'm a huge fan of `minPythonVersion` and `maxPythonVersion`. I always felt the python way of defining it as a dependency is better. Is there anything we can't use the dep for? Is this only for consistency?

**Comment by garysassano** [comment]
@mrgrain Actually, this is defined in [PEP 621](https://peps.python.org/pep-0621/) (standardized `pyproject.toml`), which Poetry hasn't adopted yet unfortunately, even though there is intention to migrate to it in the future.

The standard mandates the inclusion of a [requires-python](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#python-requires) field, which specifies the Python version required for the project.

**Comment by garysassano** [comment]
Actually, this is defined in [PEP 621](https://peps.python.org/pep-0621/) (standardized pyproject.toml), which Poetry hasn't adopted yet unfortunately, even though there is intention to migrate to it in the future.

The standard mandates the inclusion of a [requires-python](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#python-requires) field, which specifies the Python version required for the project.

**Comment by garysassano** [comment]
Actually, this is defined in [PEP 621](https://peps.python.org/pep-0621/) (standardized `pyproject.toml`), which Poetry hasn't adopted yet unfortunately, even though there is intention to migrate to it in the future.

The standard mandates the inclusion of a [requires-python](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#python-requires) field, which specifies the Python version required for the project.

**Comment by garysassano** [comment]
Actually, this is defined in [PEP 621](https://peps.python.org/pep-0621/) (standardized `pyproject.toml`), which Poetry hasn't adopted yet unfortunately, even though there is intention to migrate to it eventually.

The standard mandates the inclusion of a [requires-python](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#python-requires) field, which specifies the Python version required for the project.

**Comment by garysassano** [comment]
I've recently discovered that a method for overriding the Python version in Poetry projects was already introduced in [this PR](https://github.com/projen/projen/pull/1028). However, the approach used feels more like a workaround than a straightforward feature, and it's not exactly user-friendly (you wouldn't find out about it by browsing Projen's documentation).

In my view, it would be much more intuitive and effective to replace the current method with explicit `minPythonVersion` and `maxPythonVersion` parameters. This would make the configuration process more transparent and accessible to users.

**Comment by straygar** [comment]
My preference would be to have `formatting` and `linting` as the props, instead of `black` and `flake8` for:
- readability
- extensibility / flexibility later (to polluting high-level props with different linting packages etc.)

**Comment by garysassano** [comment]
@straygar I agree with your suggestion. In fact, I don't like how we currentloy have `poetry` and `setuptools` as high-level boolean props. Instead, I would have preferred a generic `packageManager` prop similar to the one utilized for `NodeProject`.

And yes, we could probably add a `staticTypeChecking` prop where you could use `mypy` or `pyright`.

**Comment by mrgrain** [comment]
Yep, `NodeProject` doesn't handle this great. With all of this, we should start by adding new components without integrating them into the `PythonProject` at the start.

**Comment by mikejgray** [comment]
Agreed with all of the previous comments, especially since it would make it easier to add `ruff` support later 👍🏼
